### PR TITLE
Refactors code for currently supported Python 3.6 to 3.9

### PR DESCRIPTION
**Pull request by funkyfuture**
i took the time to remove the Python 2 specific code. also i think it's easier to maintain only currently supported Python versions. there's still a lot of opportunities to improve the code and i hope to have the time to contribute more.

### CODE REVIEW

**Comment by alifhughes** [comment]
Hey @funkyfuture, thank you for the pull request! Sorry for the delay in the response, I thought I had set up notifications on this repo but it seems that this one had slipped through the net. I'm going to look into this over the next 24hours. First thing I have noticed however was the failing CI (that isn't being reported correctly.. which I'll also look into). [See here](https://travis-ci.org/github/joalla/discogs_client/jobs/737198402), seems to be running the `nosetests --with-coverage --cover-package=discogs_client` with the new versions of python that you've enabled. I can also take a look at these too.

**Comment by JKatzwinkel** [comment]
Seems like `coverage` version as specified in `requirements.txt` is just too old.  Tests succeed with e.g. `coverage>=5.0.3` or `coverage>=5.3` and Python `3.7`, `3.8` and `3.9`, but not with Python `3.6`. 

Maybe coverage could be disabled for `3.6` with something like:
```yaml
script: |
  if [ $(python -c "import sys; print(sys.version_info.minor)") -lt 7 ];
    nosetests
  else
    nosetests --with-coverage --cover-package=discogs_client
  fi
```

Also, maybe `nose` should be listed in `requirements.txt` just to be safe?

File: .travis.yml [hunk 1]
```diff
@@ -1,16 +1,12 @@
...
-    - "3.4"
-    - "3.5"
     - "3.6"
+    - "3.7"
+    - "3.8"
+    - "3.9"
 install:
     - pip install -r requirements.txt
 script:
     - nosetests --with-coverage --cover-package=discogs_client
```
Comment (JKatzwinkel) [inline .travis.yml:10]:
```suggestion
script: |
  if [ $(python -c "import sys; print(sys.version_info.minor)") -lt 7 ];
    nosetests
  else
    nosetests --with-coverage --cover-package=discogs_client
  fi
```

File: .travis.yml [hunk 2]
```diff
@@ -1,16 +1,12 @@
...
 python:
-    - "2.7"
-    - "3.4"
-    - "3.5"
     - "3.6"
+    - "3.7"
+    - "3.8"
+    - "3.9"
 install:
     - pip install -r requirements.txt
```
Comment (JKatzwinkel) [inline .travis.yml:8]:
```suggestion
install: |
    pip install -r requirements.txt
    pip install --upgrade coverage==5.0.3
```

**Comment by funkyfuture** [comment]
thank you for looking into this. i'm just not sure whether it's worth the hassle, as me thinks switching to `pytest` is imperative and should be one of the next major code updates.

@alifhughes, i have no strong opinion on this. if you want to integrate @JKatzwinkel's suggestions, i'm fine with this. (assuming your review interface enables you to do so as mine does.)

File: .travis.yml (hunk 1 above)
Comment (funkyfuture) [reply .travis.yml:10]: or `if python -c "import sys; raise SystemExit(int(sys.version_info < (3, 7)))"; then`

Comment (JKatzwinkel) [reply .travis.yml:10]: either way is ugly but the alternative is terminating python 3.6 support

**Comment by JKatzwinkel** [comment]
I don't think that switching to pytest would make coverage suddenly work with python 3.6.0 if I'm tbh.

File: .travis.yml (hunk 2 above)
Comment (alifhughes) [reply .travis.yml:8]: I think we could just specify `coverage==5.0.3` in `requirements.txt` along with `nose` as you've suggested above. Seems strange to have specific version specified here and a different one in the `requirements.txt`

**Comment by alifhughes** [comment]
Cheers for looking into the `Travis` issue @JKatzwinkel, strangely I managed to get `coverage=3.6` working with `python 3.6.10` on local machine (see pic) but doesn't mean the same for `Travis`.
Screen Shot 2020-11-15 at 2 45 58 PM

To be honest, my intention was never to have `coverage` get in the way of development, merely a measure that we can keep an eye on, so @funkyfuture I am happy for you to include @JKatzwinkel 's [suggestion here.](https://github.com/joalla/discogs_client/pull/12#discussion_r523760051). Am in agreement with @JKatzwinkel, that although moving to `pytest` would be a great improvement overall, it wouldn't fix the coverage issue on 3.6.

File: .travis.yml (hunk 2 above)
Comment (JKatzwinkel) [reply .travis.yml:8]: yes that was my intention also, I just felt that I had to leave it somewhere and I couldn't suggest it in `requirements.txt` directly without creating a fork.

**Comment by alifhughes** [comment]
Cheers for looking into the `Travis` issue @JKatzwinkel, to be honest, my intention was never to have `coverage` get in the way of development, merely a measure that we can keep an eye on, so @funkyfuture I am happy for you to include @JKatzwinkel 's [suggestion here,](https://github.com/joalla/discogs_client/pull/12#discussion_r523760051) which will disable coverage on `python 3.6`. As I mentioned [here](https://github.com/joalla/discogs_client/pull/12#discussion_r523769288) I think it would be good to bump the `coverage` version in the `requirements.txt` and include `nose` there too as @JKatzwinkel has suggested 👍 

I am in agreement with @JKatzwinkel, that although moving to `pytest` would be a **great improvement overall**, it wouldn't fix the coverage issue on 3.6. 

Am going to look into the actual changes now

File: .travis.yml (hunk 2 above)
Comment (alifhughes) [reply .travis.yml:8]: Yea no worries! I only realised that after putting that comment to as I was attempting to write it as a suggestion on here

File: .travis.yml (hunk 1 above)
Comment (alifhughes) [reply .travis.yml:10]: Agree both aren't ideal but I think I'd prefer the first as it looks _'less ugly'_

File: setup.py [hunk 3]
```diff
@@ -23,7 +23,6 @@
             ],
         install_requires=[
             'requests',
-            'six',
```
Comment (alifhughes) [inline setup.py:26]: As we've now removed the need for `six` in the project [here](https://github.com/joalla/discogs_client/pull/12/files#diff-7c36357cab15f0aaf3f77e729499050369efe7ce0915654ecc4bf88ec421da27L4). I think we can remove it from the `requirements.txt` too

**Comment by alifhughes** [comment]
Generally code changes LGTM 👍 

I've added [one minor comment](https://github.com/joalla/discogs_client/pull/12#discussion_r523773517) and also if you could apply changes to get `Travis` ignoring `python 3.6` as well, then I'm happy to merge 🎉

**Comment by JKatzwinkel** [comment]
wait what do you mean by *`Travis` ignoring `python 3.6`*!?

**Comment by alifhughes** [comment]
> wait what do you mean by _`Travis` ignoring `python 3.6`_!?

Sorry I should have been more specific! What I meant was: `Travis` to ignore/not do **`coverage`** with `python 3.6` 😄

**Review by JOJ0 — APPROVED** [review]
Thanks to all of you for these essential improvements of discogs_client! Appreciated! :-)

**Comment by alifhughes** [comment]
@funkyfuture I hope you don't mind but I took the liberty to update the `requirements.txt` [here](https://github.com/joalla/discogs_client/pull/12/commits/26558f3a143ccb6e5c487a2d3b90bdee85172eae) as @JOJ0 updated `.travis.yml`. These were the last things needed to do before we can merge 👍

**Review by alifhughes — APPROVED** [review]
Travis is now playing up, however I am going to tackle this separately, I saw that the builds were green so I am going to approve and merge. Thanks for the discussion and suggestions everyone!

**Pull request merged**